immut/sorted_set/immutable_set.mbt

```diff
@@ -581,6 +581,10 @@ pub impl[X : @quickcheck.Arbitrary + Compare] @quickcheck.Arbitrary for T[X] wit
 
 // The following are the helper functions or types used by the internal implementation of ImmutableSet
 
+///|
+/// The ratio between the sizes of the left and right subtrees.
+const BALANCE_RATIO = 5
+
```

Comment:
do you have any reference why this value is picked?

Reply:
it's copied from the sorted_map

Reply:
[Functional Pearls Efficient sets: a balancing act](https://www.cambridge.org/core/services/aop-cambridge-core/content/view/0CAA1C189B4F7C15CE9B8C02D0D4B54E/S0956796800000885a.pdf/functional_pearls_efficient_setsa_balancing_act.pdf)
I think it's from this paper